feat: add minimal support for Debian pre-dependencies

jeremy-gill opened the pull request:
Basic support for adding 'Pre-Depends:' to Debian packages added.

caarlos0 commented on nfpm.go:
```diff
@@ -303,6 +305,7 @@ func (i *Info) parseSemver() {
 type Overridables struct {
 	Replaces   []string       `yaml:"replaces,omitempty" json:"replaces,omitempty" jsonschema:"title=replaces directive,example=nfpm"`
 	Provides   []string       `yaml:"provides,omitempty" json:"provides,omitempty" jsonschema:"title=provides directive,example=nfpm"`
+	Predepends []string       `yaml:"predepends,omitempty" json:"predepends,omitempty" jsonschema:"title=predepends directive,example=nfpm"`
```
since this is only for debs, it should probably go into the specific deb options struct. (line 385)

caarlos0 reviewed (commented):
a comment with some improvement suggestions.

can you also please update the docs and add a test?

thank you for the PR :)

jeremy-gill commented:
@caarlos0, I've moved the 'predepends' tag from the platform-independent area to a Debian-specific location.  I've also tweaked `deb/deb_test.go` to provide some minimal testing of the new tag.  Finally, I updated the `www/docs/configuration.md` to explain the use of the new tag.  Is this the change you were looking for?

caarlos0 commented:
thank you, I'll add an integration test too.
